tests/operator/combine.ts

```diff
@@ -54,4 +56,20 @@ describe('Stream.prototype.combine', () => {
       },
     });
   });
+
+  it('should return a Stream when combining a MemoryStream with a Stream', (done) => {
+    const input1 = xs.periodic(50).take(4).remember();
+    const input2 = xs.periodic(80).take(3);
+    const stream: Stream<number> = input1.combine((x, y) => x + y, input2);
+    assert.strictEqual(stream instanceof Stream, true);
+    done();
+  });
+
+  it('should return a Stream when combining a MemoryStream with a MemoryStream', (done) => {
+    const input1 = xs.periodic(50).take(4).remember();
+    const input2 = xs.periodic(80).take(3).remember();
+    const stream: Stream<number> = input1.combine((x, y) => x + y, input2);
+    assert.strictEqual(stream instanceof Stream, true);
+    done();
+  });
```

Comment:
- Operators that always return `Stream` were unchanged because the operator function does explicitly "new **Stream**(producer)"
- Operators that return Stream or MemoryStream depending on the input use `new (this.ctor())(producer)` but the return type is Stream, and then MemoryStream overrides each operator to cast the return type to be MemoryStream 
- Operators that return MemoryStream always do `new MemoryStream(producer)`